.github/workflows/dekaf.yaml

```diff
@@ -0,0 +1,60 @@
+name: dekaf
+
+on:
+  push:
+    branches: [master]
+    paths:
+      - "crates/dekaf/**"
+      - "Cargo.lock"
+  pull_request:
+    branches: [master]
+    paths:
+      - ".github/workflows/dekaf.yaml"
+      - "crates/dekaf/**"
+      - "Cargo.lock"
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          submodules: false
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Login to GitHub container registry
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | \
+            docker login --username ${{ github.actor }} --password-stdin ghcr.io
+      - name: Run git describe
+        id: git_describe
+        run: echo "describe_output=$(git describe --dirty --tags)" >> $GITHUB_OUTPUT
+      # This step uses [docker/metadata-action](https://github.com/docker/metadata-action#about) to extract tags and labels that will be applied to the specified image.
+      # The `id` "meta" allows the output of this step to be referenced in a subsequent step. The `images` value provides the base name for the tags and labels.
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@a64d0487d7069df33b279515d35d60fa80e2ea62
```

Comment:
This seems to be an open point of disagreement in the github actions community. From what I could find, the strongest argument for using SHAs instead of tags is security -- avoiding a maliciously re-published tag. Beyond that, it seems people still prefer SHAs as otherwise you're relying on the action owner to correctly implement semver, or if you specify down to the patch version then you have to manually update anyway.

Github themselves [say](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/using-pre-written-building-blocks-in-your-workflow#using-shas):
> If you need more reliable versioning, you should use the SHA value associated with the version of the action. SHAs are immutable and therefore more reliable than tags or branches. However, this approach means you will not automatically receive updates for an action, including important bug fixes and security updates. 

If you still think we should stick to tags instead of SHAs, I'm happy to. Just wanted to frame the reason I went with what I did

Reply:
> However, this approach means you will not automatically receive updates for an action, including important bug fixes and security updates.

Yep, this is exactly the reason not to do this. Using fixed versions would be appropriate if and only if we were actually auditing the release and were willing to pay the cost of more intensive dependency update and re-verification cycles.

The far greater practical risk is unknowingly perpetuating a bug or vulnerability that's already been fixed for two years because we're not regularly updating dependencies.

As with my comment below, dependencies are long-term commitments, and the author of a dependency matters more when evaluating risk. `docker/` is fine. `leetHaxor/` isn't. Pinning a hash is only a temporary salve that buys immediate reproducability in exchange for deferred maintenance.

Obviously I'm addressing this question in a broader context than just GitHub actions. AFAIK the best we can do, as a community, is Cargo.toml / go.mod for expressive permissive / evergreen semver, paired with Cargo.lock / go.sum for pinned and fully reproducible builds, and automated tooling for fetching updates.

Reply:
Switched to `docker/metadata-action@v5` 👍